src/PolyType.SourceGenerator/Model/TypeShapeModel.cs

```diff
@@ -6,6 +6,10 @@ public abstract record TypeShapeModel
 {
     public required TypeId Type { get; init; }
 
+    public TypeId UnboundGenericType { get; init; }
+
+    public ReadOnlyMemory<TypeId> TypeArguments { get; init; }
```

Comment:
These types are part of the incremental model, so they need to have structural equality. Because `ROM<T>` uses reference equality, this can result in false negatives when incoming models are being compared. You can use the provided collection types that have the appropriate equality comparison:
```suggestion
    public ImmutableEquatableArray<TypeId> TypeArguments { get; init; }
```